DirectProgramming/DPC++FPGA/include/tuple.hpp

```diff
@@ -0,0 +1,119 @@
+#ifndef __TUPLE_HPP__
+#define __TUPLE_HPP__
+
+#include <type_traits>
+
+namespace fpga_tools {
+
+//
+// Generic tuple
+//
+// USAGE EXAMPLE:
+//    Tuple<char,short,int,long> my_tuple;
+//    char a = my_tuple.get<0>();
+//    short b = my_tuple.get<1>();
+//    int c = my_tuple.get<2>();
+//    long d = my_tuple.get<3>();
+//
+template <typename... Tys>
+struct Tuple {
+  Tuple(Tys... Args) : values(Args...) {}
+  Tuple() {}
+
+  //
+  // get the index'th item in the tuple of values
+  //
+  template <int index>
+  auto& get() {
```

Comment:
Naming convention for functions/member functions not consistent with rest of source code.    Is this different due to emulating a C++ standard library?

Reply:
This header file as well as the unrolled_loop.hpp (where you made the same comment below) are existing header files in the other reference designs. Therefore I did not modify them and only copied them to a common include directory. 
Maybe @mdbtucker knows why these two files do not follow the naming convention?

Reply:
Yes, the naming convention for tuple was meant to mirror std::tuple, which is why it doesn't follow the standard conventions.